run-tests.php

```diff
@@ -2574,19 +2567,20 @@ function run_test(string $php, $file, array $env): string
             $wanted_re = $temp;
 
             // Stick to basics
-            $wanted_re = str_replace('%e', '\\' . DIRECTORY_SEPARATOR, $wanted_re);
-            $wanted_re = str_replace('%s', '[^\r\n]+', $wanted_re);
-            $wanted_re = str_replace('%S', '[^\r\n]*', $wanted_re);
-            $wanted_re = str_replace('%a', '.+', $wanted_re);
-            $wanted_re = str_replace('%A', '.*', $wanted_re);
-            $wanted_re = str_replace('%w', '\s*', $wanted_re);
-            $wanted_re = str_replace('%i', '[+-]?\d+', $wanted_re);
-            $wanted_re = str_replace('%d', '\d+', $wanted_re);
-            $wanted_re = str_replace('%x', '[0-9a-fA-F]+', $wanted_re);
-            $wanted_re = str_replace('%f', '[+-]?\.?\d+\.?\d*(?:[Ee][+-]?\d+)?', $wanted_re);
-            $wanted_re = str_replace('%c', '.', $wanted_re);
-            $wanted_re = str_replace('%0', '\x00', $wanted_re);
-            // %f allows two points "-.0.0" but that is the best *simple* expression
+            $wanted_re = strtr($wanted_re, [
+                '%e' => '\\' . DIRECTORY_SEPARATOR,
+                '%s' => '[^\r\n]+',
+                '%S' => '[^\r\n]*',
+                '%a' => '.+',
+                '%A' => '.*',
+                '%w' => '\s*',
+                '%i' => '[+-]?\d+',
+                '%d' => '\d+',
+                '%x' => '[0-9a-fA-F]+',
+                '%f' => '[+-]?\.?\d+\.?\d*(?:[Ee][+-]?\d+)?', // %f allows two points "-.0.0" but that is the best *simple* expression
```

Comment:
```suggestion
                '%f' => '[+-]?(?:\d+|(?=\.))(?:\.\d+)?(?:[Ee][+-]?\d+)?',
```

Reply:
Thank you. I get that the non-capturing group and the pattern wouldn't make the expression any wrong, but given the PR is merely on code styling and readability, I think we can make this change in its own PR.

Reply:
No problem. I opened https://github.com/php/php-src/pull/8965

Reply:
Awesome, if #8965 gets merged before this one, I will make sure to rebase here. Thank you.